arduinoRemote/app/src/main/java/com/example/arduinoremote/MainActivity.java

```diff
@@ -1,55 +1,168 @@
 package com.example.arduinoremote;
 
 import android.bluetooth.BluetoothSocket;
+import android.os.Build;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
+
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
+
 import android.os.Bundle;
+import android.widget.SeekBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.IOError;
 import java.io.IOException;
 
 
 public class MainActivity extends AppCompatActivity implements View.OnClickListener {
 
-    static String instruction = "s";
-    Button forwardBut, stopBut;
+    static String instruction = "v0";
+    Button forwardBut, stopBut, reverseBut, connectBut;
+    SeekBar accelBar;
+    SeekBar steeringBar;
+    TextView accelText;
+    TextView steeringText;
     BluetoothSocket btSocket = null;
+    boolean isReversed = false;
+    ConnectBT connection;
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        if (btSocket == null) {
-            new ConnectBT().execute();
-        }
+        Log.d("Testing", "Build");
 
+        // Setting up the buttons
         forwardBut = (Button) findViewById(R.id.forwardBut);
         stopBut = (Button) findViewById(R.id.stopBut);
+        accelBar = (SeekBar) findViewById(R.id.accelBar);
+        steeringBar = (SeekBar) findViewById(R.id.steeringBar);
+        accelText = (TextView) findViewById(R.id.accelText);
+        steeringText = (TextView) findViewById(R.id.steeringText);
+        reverseBut = (Button) findViewById(R.id.reverseBut);
+        connectBut = (Button) findViewById(R.id.connectBut);
+
+        // Configuring accelBar
+        accelBar.setProgress(0);
+        accelBar.setMin(0);
+        accelBar.setMax(5);
+
+        //configuring steeringBar
+        steeringBar.setProgress(0);
+        steeringBar.setMin(-18);
+        steeringBar.setMax(18);
```

Comment:
These are all hard-coded, should they?

Reply:
We think so, but we might look into it in more detail later